Mathlib/Tactic/Cache.lean

```diff
@@ -137,12 +134,62 @@ def DeclCache.get (cache : DeclCache α) : MetaM α := do
   return a
 
 /--
-A type synonym for a `DeclCache` containing a pair of discrimination trees.
+A type synonym for a `DeclCache` containing a pair of elements.
 The first will store declarations in the current file,
 the second will store declarations from imports (and will hopefully be "read-only" after creation).
 -/
-@[reducible] def DiscrTreeCache (α : Type) : Type :=
-  DeclCache (DiscrTree α true × DiscrTree α true)
+def DeclCache2 (α : Type) := DeclCache (α × α)
+
+instance (α : Type) [h : Nonempty α] : Nonempty (DeclCache2 α) :=
+  h.elim fun x => @instNonemptyDeclCache _ ⟨x,x⟩
```

Comment:
Is the issue that the `Prod.nonempty` instance is not in scope? Can you add a comment to that effect?

Reply:
Indeed, that is the case, thanks! Why is that instance not further up the dependency chain?